hera_notebook_templates/cli.py

```diff
@@ -70,10 +69,14 @@ def runfunc(ctx, basename, **kwargs):
         output_path = Path(ctx.obj['output_dir']) / f"{basename}.ipynb"
 
         if ctx.obj['toml'] is not None:
-            toml = toml.load(ctx.obj['toml'])
+            cfg = toml.load(ctx.obj['toml'])
             if ctx.obj['toml_section'] is not None:
-                toml = toml[ctx.obj['toml_section']]
-            kwargs.update(toml)
+                cfg = cfg[ctx.obj['toml_section']]
+
+            for name in cfg:
+                if name not in kwargs:
+                    raise ValueError(f"Parameter '{name}' not found in notebook parameters.")
```

Comment:
Are we certain we want this? I'm thinking about the case where parameters are shared between notebooks, so in the toml the `args` previously referenced stuff from multiple sections. For a larger toml with many steps, if we move some or all steps to this interface, I think this'll break if we pass in a whole toml but only use part of it.

Reply:
I added an option to each of the commands `--extra-param-behaviour` that switches between raising, warning and ignoring such extra parameters. Also added a comment as to why it might be useful.